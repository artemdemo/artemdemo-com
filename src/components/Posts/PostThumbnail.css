.post-thumbnail {
  border-radius: 5px;
}
