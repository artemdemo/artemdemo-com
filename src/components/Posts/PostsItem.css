/*
    See explanation in the code PostsItem.jsx
*/
code {
    padding: .1em;
    border-radius: .3em;
    color: black;
    background: #f5f2f0;
    text-shadow: 0 1px white;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
}