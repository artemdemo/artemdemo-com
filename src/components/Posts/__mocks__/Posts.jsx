import React from 'react';

const Posts = props => (
    <div data-mock='Posts'>
        {props.children}
    </div>
);

export default Posts;