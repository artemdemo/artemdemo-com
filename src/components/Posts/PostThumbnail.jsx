import React from 'react';
import PropTypes from 'prop-types';
import { GatsbyImage, getImage } from 'gatsby-plugin-image';
import './PostThumbnail.css';

export const PostThumbnail = ({ featuredImage }) => {
  if (featuredImage) {
    return (
      <GatsbyImage
        className="post-thumbnail"
        image={getImage(featuredImage)}
        alt=""
      />
    );
  }
  return null;
};

PostThumbnail.propTypes = {
  featuredImage: PropTypes.shape({}),
};
