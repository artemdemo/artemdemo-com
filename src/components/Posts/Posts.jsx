import React from 'react';
import PropTypes from 'prop-types';
import PostsItem from './PostsItem';

const Posts = (props) => {
  const {list} = props;

  return (
    <React.Fragment>
      {list.map((post) => {
        const {node} = post;
        return (
          <PostsItem
            node={node}
            key={node.fields.slug}
          />
        )
      })}
    </React.Fragment>
  );
};

Posts.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({})),
};

Posts.defaultProps = {
  list: [],
};

export default Posts;
