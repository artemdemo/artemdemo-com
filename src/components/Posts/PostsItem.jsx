import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'gatsby';
import _get from 'lodash/get';
import PostDate from '../PostDate/PostDate';
import * as utils from '../../services/utils';
import {renderMd} from '../../services/md';

// Yes, I know, not the best solution.
// The problem is that I need to use here custom service for markdown,
// because `gatsby-transformer-remark` will work only for posts, but _not_ for excerpts.
import './PostsItem.css';

class PostsItem extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      content: null,
    };
  }

  componentDidMount() {
    const excerpt = _get(this.props, 'node.excerpt');

    if (excerpt) {
      renderMd(excerpt)
        .then(content => this.setState({content}));
    }
  }

  render() {
    const {node} = this.props;
    if (!node) {
      return null;
    }
    const title = _get(node, 'frontmatter.title') || node.fields.slug;
    return (
      <div className='post-item'>
        <h2>
          <Link to={utils.prefixBlog(node.fields.slug)}>
            {title}
          </Link>
        </h2>
        <PostDate>
          {node.frontmatter.date}
        </PostDate>
        <p dangerouslySetInnerHTML={{__html: this.state.content}}/>
      </div>
    );
  }
}

PostsItem.propTypes = {
  node: PropTypes.shape({}),
};

PostsItem.defaultProps = {
  node: null,
};

export default PostsItem;
