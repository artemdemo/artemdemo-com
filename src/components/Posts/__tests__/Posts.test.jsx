import React from 'react';
import renderer from 'react-test-renderer';
import Posts from '../Posts';

jest.mock('../PostsItem');

describe('Posts', () => {
    it('should render empty', () => {
        const tree = renderer.create(
            <Posts />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with posts', () => {
        const tree = renderer.create(
            <Posts
                list={[
                    {
                        node: {
                            fields: {slug: '/first'},
                        },
                    },
                    {
                        node: {
                            fields: {slug: '/second'},
                        },
                    }
                ]}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});