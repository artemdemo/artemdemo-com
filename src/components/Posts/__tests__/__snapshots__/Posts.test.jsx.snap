// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts should render empty 1`] = `null`;

exports[`Posts should render with posts 1`] = `
Array [
  <div
    data-mock="PostsItem"
  >
    {
  "node": {
    "fields": {
      "slug": "/first"
    }
  }
}
  </div>,
  <div
    data-mock="PostsItem"
  >
    {
  "node": {
    "fields": {
      "slug": "/second"
    }
  }
}
  </div>,
]
`;
